crates/solver/src/solver/score_computation.rs

```diff
@@ -51,7 +51,16 @@ impl ScoreCalculator {
             - self.gas_price_factor * gas_price / 10_000_000_000.
             - self.nmb_orders_factor * nmb_orders as f64;
         let success_probability = 1. / (1. + exponent.exp());
-        let score = success_probability * (surplus + fees - gas_amount * gas_price);
+        let obj = surplus + fees - gas_amount * gas_price;
+        let cap = 10000000000000000.0;
+        let score = if success_probability * obj <= cap && (1.0 - success_probability) * obj <= cap
```

Comment:
I'm not able to verify that this logic makes sense without an explanation. 😅 
A minor improvement to readability could be to add
```rust
let revert_probability = 1.0 - success_probability;
```
and use that inside the conditions and calculations.